pynecone/state.py

```diff
@@ -730,11 +735,18 @@ def get_delta(self) -> Delta:
         """
         delta = {}
 
-        # Return the dirty vars and dependent computed vars
-        self._mark_dirty_computed_vars()
-        delta_vars = self.dirty_vars.intersection(self.base_vars).union(
-            self._dirty_computed_vars()
+        # Apply dirty variables down into substates
+        self.dirty_vars.update(self._always_dirty_computed_vars())
+        self.mark_dirty()
+
+        # Return the dirty vars for this instance, any cached/dependent computed vars,
+        # and always dirty computed vars (cache=False)
+        delta_vars = (
+            self.dirty_vars.intersection(self.base_vars)
+            .union(self._dirty_computed_vars())
+            .union(self._always_dirty_computed_vars())
```

Comment:
Since you updated the `self.dirty_vars` already on line 739, do we need to add the `always_dirty_computed_vars()` again here?

Reply:
Yes because delta vars starts by intersect dirty_vars with base_vars, which does NOT include computed vars. So then we union back in the dirty computed vars that belong to this instance (not inherited)